src/Fees.sol

```diff
@@ -10,68 +10,23 @@ library PDPFees {
-    /// @param estimatedGasFee The estimated gas fee in AttoFIL.
-    /// @param filUsdPrice The price of FIL in USD.
-    /// @param filUsdPriceExpo The exponent of the price of FIL in USD.
+    /// @notice Calculates the proof fee based on the dataset size and a provided per-TiB fee.
     /// @param rawSize The raw size of the proof in bytes.
-    /// @param nProofEpochs The number of proof epochs.
+    /// @param feePerTiB The fee rate per TiB in AttoFIL (source of truth lives in PDPVerifier).
     /// @return proof fee in AttoFIL
-    /// @dev The proof fee is calculated based on the gas fee and the raw size of the proof
-    /// The fee is 1% of the projected reward and is reduced in the case gas cost of proving is too high.
-    function proofFeeWithGasFeeBound(
-        uint256 estimatedGasFee, // in AttoFIL
-        uint64 filUsdPrice,
-        int32 filUsdPriceExpo,
-        uint256 rawSize,
-        uint256 nProofEpochs
-    ) internal view returns (uint256) {
-        require(
-            estimatedGasFee > 0 || block.basefee == 0, "failed to validate: estimated gas fee must be greater than 0"
-        );
-        require(filUsdPrice > 0, "failed to validate: AttoFIL price must be greater than 0");
+    /// @dev The proof fee is calculated as: fee_perTiB * datasetSize_in_TiB
+    function calculateProofFee(uint256 rawSize, uint256 feePerTiB) internal pure returns (uint256) {
         require(rawSize > 0, "failed to validate: raw size must be greater than 0");
 
-        // Calculate reward per epoch per byte (in AttoFIL)
-        uint256 rewardPerEpochPerByte;
-        if (filUsdPriceExpo >= 0) {
-            rewardPerEpochPerByte = (ESTIMATED_MONTHLY_TIB_STORAGE_REWARD_USD * FIL_TO_ATTO_FIL)
-                / (TIB_IN_BYTES * EPOCHS_PER_MONTH * filUsdPrice * (10 ** uint32(filUsdPriceExpo)));
-        } else {
-            rewardPerEpochPerByte = (
-                ESTIMATED_MONTHLY_TIB_STORAGE_REWARD_USD * FIL_TO_ATTO_FIL * (10 ** uint32(-filUsdPriceExpo))
-            ) / (TIB_IN_BYTES * EPOCHS_PER_MONTH * filUsdPrice);
-        }
-
-        // Calculate total reward for the proving period
-        uint256 estimatedCurrentReward = rewardPerEpochPerByte * nProofEpochs * rawSize;
-
-        // Calculate gas limits
-        uint256 gasLimitRight = (estimatedCurrentReward * GAS_LIMIT_RIGHT_PERCENTAGE) / 100;
-        uint256 gasLimitLeft = (estimatedCurrentReward * GAS_LIMIT_LEFT_PERCENTAGE) / 100;
-
-        if (estimatedGasFee >= gasLimitRight) {
-            return 0; // No proof fee if gas fee is above right limit
-        } else if (estimatedGasFee >= gasLimitLeft) {
-            return gasLimitRight - estimatedGasFee; // Partial discount on proof fee
-        } else {
-            return (estimatedCurrentReward * PROOF_FEE_PERCENTAGE) / 100;
-        }
+        // Calculate fee as: feePerTiB * (rawSize / TIB_IN_BYTES)
+        return (feePerTiB * rawSize) / TIB_IN_BYTES;
```

Comment:
since the fee is much larger than 2 ** 40, can just denominate fee per byte in attofil

Reply:
happens to be 209.18 attofil per byte

if we want the precision we can do attofil per kbyte

but can keep this as-is and do `>>40` instead of division by `2**40`